admin/spec/service/check-window.service.spec.js

```diff
@@ -189,5 +189,45 @@ describe('check-window.service', () => {
       expect(result.toString).toBe(expected.toString)
       expect(result).toBeTruthy()
     })
+    describe('#isLoginAllowed', () => {
+      it('should allow pupil to login if within range', async (done) => {
+        spyOn(checkWindowDataService, 'sqlFindOneCurrent').and.returnValue({
+          checkStartDate: moment.utc().subtract(1, 'days'),
+          checkEndDate: moment.utc().add(1, 'days')
+        })
+        try {
+          await service.isLoginAllowed()
+        } catch (error) {
+          fail('not expected to throw')
+        }
+        done()
+      })
+      it('should disallow pupil to login if checkStartDate is in the future', async (done) => {
+        spyOn(checkWindowDataService, 'sqlFindOneCurrent').and.returnValue({
+          checkStartDate: moment.utc().add(1, 'days'),
+          checkEndDate: moment.utc().add(2, 'days')
+        })
+        try {
+          await service.isLoginAllowed()
+          fail('not expected to throw')
+        } catch (error) {
+          expect(error.message).toBe('Pupil not allowed to log in')
+        }
+        done()
+      })
+      it('should disallow pupil to login if checkEndDate is in the past', async (done) => {
+        spyOn(checkWindowDataService, 'sqlFindOneCurrent').and.returnValue({
+          checkStartDate: moment.utc().subtract(3, 'days'),
+          checkEndDate: moment.utc().subtract(2, 'days')
+        })
+        try {
+          await service.isLoginAllowed()
+          fail('not expected to throw')
+        } catch (error) {
+          expect(error.message).toBe('Pupil not allowed to log in')
```

Comment:
i think we have misunderstood the discussion here - the point of this extra work was to indicate why they cannot login.

so instead of just `Pupil not allowed to log in` can we indicate....
- `There is no active check window`
- `The active check window has expired`
- `The active check window has not started`
These actual messges should be checked with the BAs.  if they dont want them shown to the pupil, at least store them via winston so when debugging you have a clear indication of why they cannot login

Reply:
@GuyHarwood the code you were commenting on is outdated. If you check the latest commit I made yesterday there are tests to indicate `The active check window has expired` and `The active check window has not started`.
The only part that is missing is detection of no active check window which I will cover now.

Reply:
strange, i clicked on 'view changes'.

thanks for the heads up, looks much better 👍

Reply:
Yea conversation tab can be misleading sometimes in terms of latest code state.
Btw updated